src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/MovementResourceTest.kt

```diff
@@ -0,0 +1,418 @@
+      LocalDateTime.of(2018, 4, 25, 0, 0, 0).truncatedTo(ChronoUnit.DAYS).format(DateTimeFormatter.ISO_LOCAL_DATE_TIME),
+      LocalDate.of(2018, 5, 1).format(DateTimeFormatter.ISO_LOCAL_DATE),
+    )
+    assertThatStatus(response, 200)
+    assertThatJson(response.body).isEqualTo("movements_on_day.json".readFile())
+  }
+
+  @Test
+  fun testReadRollcountByAgency() {
+    val token = authTokenHelper.getToken(AuthToken.NORMAL_USER)
+    val response = testRestTemplate.exchange(
+      "/api/movements/rollcount/{agencyId}/movements",
+      GET,
+      createHttpEntity(token, null),
+      object : ParameterizedTypeReference<String>() {},
+      "LEI",
+    )
+    assertThatStatus(response, 200)
+    assertThatJson(response.body).isEqualTo("{\"in\":0,\"out\":0}")
+  }
+
+  @Test
+  fun testReadTodaysMovementsByAgencyEnRoute() {
+    val token = authTokenHelper.getToken(AuthToken.NORMAL_USER)
+    val response = testRestTemplate.exchange(
+      "/api/movements/{agencyId}/enroute",
+      GET,
+      createHttpEntity(token, null),
+      object : ParameterizedTypeReference<String>() {},
+      "LEI",
+    )
+    assertThatStatus(response, 200)
+    assertThatJson(response.body).isEqualTo(
+      """
+      [{
+        "offenderNo": "A1183SH",
+          "bookingId":-44,
+          "dateOfBirth": "1980-01-02",
+          "firstName": "SAM",
+          "lastName": "HEMP",
+          "fromAgency": "BMI",
+          "fromAgencyDescription": "Birmingham",
+          "toAgency": "LEI",
+          "toAgencyDescription": "Leeds",
+          "movementType": "TRN",
+          "movementTypeDescription": "Transfers",
+          "movementReason": "NOTR",
+          "movementReasonDescription": "Normal Transfer",
+          "directionCode": "OUT",
+          "movementTime": "13:00:00",
+          "movementDate": "2017-10-12"
+        },
+        {
+          "offenderNo": "A1183AD",
+          "bookingId":-45,
+          "dateOfBirth": "1980-01-02",
+          "firstName": "AMY",
+          "lastName": "DENTON",
+          "fromAgency": "BMI",
+          "fromAgencyDescription": "Birmingham",
+          "toAgency": "LEI",
+          "toAgencyDescription": "Leeds",
+          "movementType": "TRN",
+          "movementTypeDescription": "Transfers",
+          "movementReason": "NOTR",
+          "movementReasonDescription": "Normal Transfer",
+          "directionCode": "OUT",
+          "movementTime": "15:00:00",
+          "movementDate": "2017-10-12"
+        }]
+    """,
+    )
+  }
+
+  @Test
+  fun testGetRolllcountByAgencyEnroute() {
+    val token = authTokenHelper.getToken(AuthToken.NORMAL_USER)
+    val response = testRestTemplate.exchange(
+      "/api/movements/rollcount/{agencyId}/enroute",
+      GET,
+      createHttpEntity(token, null),
+      object : ParameterizedTypeReference<String>() {},
+      "LEI",
+    )
+    assertThatStatus(response, 200)
+    assertThatJson(response.body).isEqualTo("2")
+  }
+
+  @Nested
```

Comment:
converted to kotlin and only added authorisation tests for  `/api/movements/{agencyId}/in/{isoDate}` and  `/api/movements/rollcount/{agencyId}/in-reception` - additional tests exist as cucumber tests, these check the actual values returned